core/src/main/scala/io/chrisdavenport/shellfish/FilesOs.scala

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (c) 2024 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.chrisdavenport.shellfish
+
+import cats.syntax.all.*
+import cats.effect.IO
+
+import fs2.{Stream, Chunk}
+import fs2.interop.scodec.*
+import fs2.io.file.{Files, Path, Flags}
+
+import scodec.{Codec, codecs}
+import scodec.bits.ByteVector
+
+object FilesOs {
```

Comment:
bike shedding here, I quite like `os-lib` style that looks like the image below, I wonder could and should We do something similar?

<img width="362" alt="Screenshot 2024-06-20 at 10 07 00" src="https://github.com/davenverse/shellfish/assets/437967/0e0f5f32-db3a-4712-a747-971db261e97e">

Reply:
This is a difficult one. I personally don't like a lot this style of calling apply methods over and over but I had to admit it is pretty clever. We should discuss if we should keep the pruposed by Tonio, this one or maybe both?

Reply:
agree, this one is difficult one. I'm not sure either. A good thing with this approach is discoverability. users can discover everything by typing `os.` ide will show all the options.

Reply:
Hmm, I'm not sure. I really don't like this style too, but we should at ergonomics for users and if our priority is to make people transition from os-lib to this lib as smoothly as possible. IMHO, as this is not the main goal of the lib, there's not the need to introduce such a convoluted structure.

Pinging @armanbilge and all his winsdom.